src/Shared/Constants.cs

```diff
@@ -82,13 +82,7 @@ internal static class MSBuildConstants
         /// <summary>
         /// Current version of this MSBuild Engine assembly in the form, e.g, "12.0"
         /// </summary>
-        internal static string CurrentProductVersion
-        {
-            get
-            {
-                return "15.0";
-            }
-        }
+        internal static string CurrentProductVersion => "16.0";
```

Comment:
Would it be better if this got read from the dll informational version major part?